File: src/main/java/syntax/java/Bicho.java
package syntax.java;

class Nomeado {
	String nome;
	public Nomeado comNome(String nome){
		this.nome = nome;
		return this;
	}
}

class Bicho extends Nomeado {
	public void oi() {
		System.out.println(nome);
	}
	
	public static void main(String[] args) {
		new Bicho().comNome("Rex");//.oi();
	}
}

File: src/main/scala/syntax/scala/Bicho.scala
package syntax.scala
import scala.language.postfixOps

object Bicho extends App{

  trait Nomeado {
    var name: String = _

    def comNome(name: String): this.type = {
      this.name = name
      this
    }
  }
  
  class Pessoa extends Nomeado {
    def oi = println(s"Oi $name")
  }
  
  new Pessoa() comNome("Julio") oi 
}